doc/ref_arch/kubernetes/chapters/chapter06.md

```diff
@@ -10,9 +10,56 @@
+- Containers: A lightweight and portable executable image that contains software and all of its dependencies. The containers need to be prevented from accessing the underlying OS like loading of kernel modules, mounting of directories of underlying OS etc and it must also be ensured that they don't run in priveleged mode..
+- Pods: A Pod represents a set of running containers on your cluster. Kuberenetes inherently offers pod security policies that define a set of conditions that a pod must run with in order to be accepted into the system. These policies help in ensuring the necessary checks for running the pods.
+- Kubernetes Node: A Kubernetes node in an unsecured boundary can lead to a potential threat to the running workloads. Such a node may be hardened in terms of security by disabling unused ports, prohibiting root access etc.
+- Kubernetes Master: A master node in an unsecured boundary can lead to a potential threat to the running workloads. A master may be hardened in terms of security by disabling unused ports, prohibiting root access etc.
+- Kubernetes Control Plane: The container orchestration layer that exposes the API and interfaces to define, deploy, and manage the lifecycle of containers. The communication over these APIs needs to be secured via different mechanisms like TLS encryption, API authentication via LDAP etc.
+
+<a name="6.4"></a>
+## 6.4 Isolation
+### VM vs. Container Isolation
+Sometimes container isolation is compared directly with VM based isolation, with the conclusion '*there are issues with container isolation, it is not as good as VM isolation*'. Such 1:1 comparison is not reasonable because VM and container based isolation are fundamentally different:
+- VMs: hard isolation, in the layers underlying the application SW
+- Containers: isolation by SW based mechanisms available in OS, Docker and Kubernetes. A container workload is just a set of Linux processes. It is _possible_ to configure SW based _additional isolation_ for container workloads, for example by kernel namespaces.
+
+Thus the primary isolation mechanism in Kubernetes environment should be VM or physical machine based isolation. This means: multiple container applications should not be deployed together in the same Kubernetes cluster - unless those have been planned and verified to co-exist.
+
+### Container Isolation in Kubernetes Cluster
+#### Namespaces  
+Kernel namespaces should be used to provide process level isolation within a Kubernetes cluster. There are different types of kernel namespaces like PID and network. The default is to allocate one namespace per container application, in case several applications can be deployed in the same cluster.
```

Comment:
Could it be further clarified when kernel namespaces vs kubernetes namespaces is meant, and whether container application refers to CNF (instead of containerized application like postgresql)?

Reply:
Resolved